pkg/recovery/recover.go

```diff
@@ -0,0 +1,370 @@
+		return nil, err
+	} else {
+		as = append(as, objAs...)
+	}
+	if objAs, err := serializeListObjToYAML(&cp.secrets); err != nil {
+		return nil, err
+	} else {
+		as = append(as, objAs...)
+	}
+	if objAs, err := serializeListObjToYAML(&cp.services); err != nil {
+		return nil, err
+	} else {
+		as = append(as, objAs...)
+	}
+	if objAs, err := serializeListObjToYAML(&cp.serviceAccounts); err != nil {
+		return nil, err
+	} else {
+		as = append(as, objAs...)
+	}
+	return as, nil
+}
+
+// renderBootstrap returns assets for a bootstrap control plane that can be used with `bootkube
+// start` to re-bootstrap a control plane. These assets are derived from the self-hosted control
+// plane that was recovered by the backend, but modified for direct injection into a kubelet.
+func (cp *controlPlane) renderBootstrap(kubeConfigPath string) (asset.Assets, error) {
+	pods, err := extractBootstrapPods(cp.daemonSets.Items, cp.deployments.Items)
+	if err != nil {
+		return nil, err
+	}
+	requiredSecrets, err := fixUpBootstrapPods(pods)
+	if err != nil {
+		return nil, err
+	}
+	as, err := outputBootstrapPods(pods)
+	if err != nil {
+		return nil, err
+	}
+	secrets, err := outputBootstrapSecrets(cp.secrets.Items, requiredSecrets)
+	if err != nil {
+		return nil, err
+	}
+	as = append(as, secrets...)
+	return as, nil
+}
+
+// extractBootstrapPods extracts bootstrap pod specs from daemonsets and deployments.
+func extractBootstrapPods(daemonSets []v1beta1.DaemonSet, deployments []v1beta1.Deployment) ([]v1.Pod, error) {
+	var pods []v1.Pod
+	for _, ds := range daemonSets {
+		if componentName := ds.Labels["component"]; bootstrapComponents[componentName] {
+			pod := v1.Pod{Spec: ds.Spec.Template.Spec}
+			if err := setBootstrapPodMetadata(&pod, ds.ObjectMeta); err != nil {
+				return nil, err
+			}
+			pods = append(pods, pod)
+		}
+	}
+	for _, ds := range deployments {
+		if componentName := ds.Labels["component"]; bootstrapComponents[componentName] {
+			pod := v1.Pod{Spec: ds.Spec.Template.Spec}
+			if err := setBootstrapPodMetadata(&pod, ds.ObjectMeta); err != nil {
+				return nil, err
+			}
+			pods = append(pods, pod)
+		}
+	}
+	return pods, nil
+}
+
+// setBootstrapPodMetadata creates valid metadata for a bootstrap pod. Currently it sets the
+// TypeMeta and Name, Namespace, and Annotations on the ObjectMeta.
+func setBootstrapPodMetadata(pod *v1.Pod, parent metav1.ObjectMeta) error {
+	if err := setTypeMeta(pod); err != nil {
+		return err
+	}
+	pod.ObjectMeta = metav1.ObjectMeta{
+		Annotations: parent.Annotations,
+		Name:        "bootstrap-" + parent.Name,
+		Namespace:   parent.Namespace,
+	}
+	return nil
+}
+
+// fixUpBootstrapPods modifies extracted bootstrap pod specs to have correct metadata and
+// point to filesystem-mount-based secrets. It returns a set of secrets that must also be rendered
+// in order for the bootstrap pods to be functional.
+// TODO(diegs): also output the set of reqiured configMaps.
```

Comment:
SG. I was going to go ahead and add it but it's a bit of work, will follow up immediately after this PR.